hydra-tui/src/Hydra/TUI.hs

```diff
@@ -172,22 +177,37 @@ warn :: Text -> State -> State
 warn = report Error
 
 report :: Severity -> Text -> State -> State
-report typ msg = feedbackL ?~ UserFeedback typ msg
+report typ msg st = case st of
+  Connected{me, nodeHost, peers, headState, dialogState, feedbacks, now} ->
+    Connected{me, nodeHost, peers, headState, dialogState, feedbacks = userFeedback : feedbacks, now}
+   where
+    userFeedback = UserFeedback typ msg now
+  disconnected -> disconnected
 
 --
 -- Update
 --
+clearFeedbacks :: UTCTime -> UTCTime -> [UserFeedback] -> [UserFeedback]
+clearFeedbacks startUpTime now feedbacks =
+  if now > startUpTime then nonExpiredFeedbacks else mempty
+ where
+  expiredFeedback UserFeedback{time} = 3 < diffUTCTime now time
+  nonExpiredFeedbacks = filter (not . expiredFeedback) feedbacks
 
-clearFeedback :: State -> State
-clearFeedback = feedbackL .~ empty
+clearState :: UTCTime -> State -> State
+clearState startUpTime = \case
+  s@Connected{feedbacks, now} ->
+    s & feedbacksL .~ clearFeedbacks startUpTime now feedbacks
+  disconnected -> disconnected
```

Comment:
Instead of the sledgehammer (calling `clearState` on each event update), I would suggest to clear feedback in the state to keep things simple (and maybe we want the scrollable full log eventually?).

So instead of this, we could just don't render things if they are too old? WDYT?

Reply:
I don't know if I am understanding this request correctly.

What I can say is that:
- currently `clearState` updates the state and clears any "old" feedback.
- I've tried to do this clear during `drawFeedback` but the UI gets buggy when you update the state there.
- I've also tried to do this clear on every Tick instead of every event.
- Since I saw the previous approach was doing the clear for each event, I thought there was some rationale behind so I left the same semantics.

Please, can you elaborate more on how you see this request for change?
How can we push this clearing to the state?
Is it by defining some custom Lens magic?

Looking forward to your feedback! 🤗

Reply:
Oh I'm so sorry.. it should have read "I would suggest to NOT clear feedback in the state to keep things simple".